source/tls_channel_handler.c

```diff
@@ -146,16 +115,29 @@ int aws_tls_ctx_options_init_client_mtls_from_path(
     options->max_fragment_size = g_aws_channel_max_fragment_size;
 
     if (aws_byte_buf_init_from_file(&options->certificate, allocator, cert_path)) {
-        return AWS_OP_ERR;
+        goto error;
+    }
+
+    if (aws_sanitize_pem(&options->certificate, allocator)) {
+        AWS_LOGF_ERROR(AWS_LS_IO_TLS, "static: Invalid certificate. File must contain PEM encoded data");
+        goto error;
     }
 
     if (aws_byte_buf_init_from_file(&options->private_key, allocator, pkey_path)) {
-        aws_byte_buf_clean_up(&options->certificate);
-        return AWS_OP_ERR;
+        goto error;
     }
-    s_tls_ctx_options_pem_sanitize(options);
+
+    if (aws_sanitize_pem(&options->private_key, allocator)) {
+        AWS_LOGF_ERROR(AWS_LS_IO_TLS, "static: Invalid private key. File must contain PEM encoded data");
+        goto error;
+    }
+
     return AWS_OP_SUCCESS;
+error:
+    aws_tls_ctx_options_clean_up(options);
```

Comment:
I went ahead and verified that this was actually safe, and it is, but I'm wondering are there other places in the whole code base that should be calling cleanup on the init object on a failure.  I have it in my mind that we do that for new/delete/release creation patterns but was surprised by this.  It makes sense but requires a zero-struct at the end of clean up.

Reply:
our error handling patterns are not consistent through the whole codebase. I've definitely settled on this as the preferred strategy, which is why I deployed it here.

Yeah, our clean_up() functions are supposed to be idempotent, and zero-struct at the end of clean_up() really helps with that.